ksql-engine/src/main/java/io/confluent/ksql/connect/Connectors.java

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright 2019 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License (the "License"; you may not use
+ * this file except in compliance with the License. You may obtain a copy of the
+ * License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.ksql.connect;
+
+import com.google.common.base.Splitter;
+import io.confluent.ksql.metastore.model.DataSource.DataSourceType;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import org.apache.kafka.connect.runtime.rest.entities.ConnectorInfo;
+
+final class Connectors {
+
+  static final String CONNECTOR_CLASS = "connector.class";
+  static final String JDBC_SOURCE_CLASS = "io.confluent.connect.jdbc.JdbcSourceConnector";
+
+  private Connectors() {
+  }
+
+  static Optional<Connector> fromConnectInfo(final ConnectorInfo connectorInfo) {
+    return fromConnectInfo(connectorInfo.config());
+  }
+
+  @SuppressWarnings("SwitchStatementWithTooFewBranches") // will soon expand to more
+  static Optional<Connector> fromConnectInfo(final Map<String, String> properties) {
+    final String clazz = properties.get(CONNECTOR_CLASS);
+    if (clazz == null) {
+      return Optional.empty();
+    }
+
+    switch (clazz) {
+      case JDBC_SOURCE_CLASS:
+        return Optional.of(jdbc(properties));
+      default:
+        return Optional.empty();
+    }
+  }
+
+  private static Connector jdbc(final Map<String, String> properties) {
```

Comment:
let's handle this problem as it comes. I was considering the best way to do it and I'm not sure that having a class per Connector will make it any cleaner - in fact I think it might make the amount of boilerplate annoying to refactor. I may be wrong, but refactoring in the future is easy :)